kaitai_utils.py

```diff
@@ -101,28 +130,31 @@ def _object_to_dict(self, instance: Any, path: str) -> Generator[Dict[str, Any],
             elif _is_list(value_object):
                 yield _to_lower_camel_case(key), self._list_to_dict(value_object, path)
             elif isinstance(value_object, bytes):
-                yield self.process_byte_array(value_object, path, key)
+                if len(value_object) > self.max_byte_array_length:
+                    if len(value_object) < hansken_extraction_plugin.runtime.constants.MAX_CHUNK_SIZE:
+                        length = new_debug[key]['end'] - new_debug[key]['start']
+                        child_builder = self.trace.child_builder(path)
+                        if self.has_process:
+                            child_builder.update(data={'raw': value_object}).build()
+                            yield _to_lower_camel_case(key), f'data block of size: {len(value_object)} (stored as {path})'
+                        else:
+                            child_builder.add_transformation('raw', RangedTransformation(
+                            [Range(new_debug[key]['start'], length)]))
+                            child_builder.build()
+                            yield _to_lower_camel_case(key), f'data block of size: {len(value_object)} (stored as {path})'
+                    else:
+                        yield _to_lower_camel_case(key), f'data block of size: {len(value_object)} (not added as child trace because size exceeds MAX_CHUNK_SIZE)'
+                else:
+                    yield _to_lower_camel_case(key), value_object.hex()
             else:
                 yield _to_lower_camel_case(key), _process_value(value_object)
 
-    def process_byte_array(self, value_object: bytes, path: str, key: str):
```

Comment:
What is the reason that this method has now been inlined above?

I find this previous code a bit better. If possible, please extract a method again.